nncf/accuracy_aware_training/algo.py

```diff
@@ -0,0 +1,316 @@
+"""
+ Copyright (c) 2019-2020 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import TypeVar
+import os.path as osp
+from abc import ABC, abstractmethod
+from copy import copy
+from shutil import copyfile
+
+import numpy as np
+import torch
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+from torch.utils.tensorboard import SummaryWriter
+
+import nncf.accuracy_aware_training.restricted_pickle_module as restricted_pickle_module
+from nncf.initialization import TrainEpochArgs
+from nncf.compression_method_api import PTStubCompressionScheduler
+from nncf.compression_method_api import CompressionLevel
+from nncf.common.utils.logger import logger as nncf_logger
+from nncf.common.utils.registry import Registry
+from nncf.api.composite_compression import CompositeCompressionAlgorithmController
+from nncf.api.composite_compression import CompositeCompressionScheduler
+from nncf.accuracy_aware_training.utils import is_main_process, print_statistics, configure_paths
+from nncf.checkpoint_loading import load_state
+
+
+ModelType = TypeVar('ModelType')
+ACCURACY_AWARE_CONTROLLERS = Registry('accuracy_aware_controllers')
+
+
+class TrainingRunner(ABC):
```

Comment:
Could you move the abstract class to the common module?

Reply:
done